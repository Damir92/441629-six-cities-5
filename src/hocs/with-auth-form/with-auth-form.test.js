import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

import withAuthForm from './with-auth-form';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <Fragment>
      {children}
    </Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAuthForm(MockComponent);

describe(`with-auth-form hoc rendered correctly`, () => {
  it(`withAuthForm without props rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped>
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`withAuthForm with props rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        email={`test@test.js`}
        password={`123`}
      >
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
