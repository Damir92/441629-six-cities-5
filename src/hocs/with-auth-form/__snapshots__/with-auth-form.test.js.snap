// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-auth-form hoc rendered correctly withAuthForm with props rendered correctly 1`] = `null`;

exports[`with-auth-form hoc rendered correctly withAuthForm without props rendered correctly 1`] = `null`;
